### PR TITLE
Minimal browser test with puppeteer.

### PR DESCRIPTION
**Pull request by isoos**
#2944

### CODE REVIEW

File: pkg/pub_integration/lib/src/headless_env.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:puppeteer/puppeteer.dart';
+// ignore: implementation_imports
+import 'package:puppeteer/src/page/page.dart' show ClientError;
+
+class HeadlessEnv {
```
Comment (jonasfj) [inline pkg/pub_integration/lib/src/headless_env.dart]: Can we give this a online description?

File: .travis.yml [hunk 2]
```diff
@@ -5,6 +5,12 @@ language: dart
 services:
   - "redis-server"
 
+addons:
+  chrome: stable
+env:
+  global:
+    - CHROME_BIN=/usr/bin/google-chrome
```
Comment (jonasfj) [inline .travis.yml]: Let's detect this in our test framework...

File: pkg/pub_integration/lib/src/headless_env.dart [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  final clientErrors = <ClientError>[];
+
+  HeadlessEnv({Directory tempDir})
+      : tempDir =
+            tempDir ?? Directory.systemTemp.createTempSync('pub-headless');
+
+  Future _startBrowser() async {
+    if (_browser != null) return;
+    final chromeBin = Platform.environment['CHROME_BIN'];
+    if (chromeBin != null && !File(chromeBin).existsSync()) {
```
Comment (jonasfj) [inline pkg/pub_integration/lib/src/headless_env.dart]: Can we please do some bare minimum for automatic resolution, I think we can just search `PATH` or use `google-chrome` if we don't need a absolute path.

File: pkg/pub_integration/lib/src/headless_env.dart [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      throw StateError(
+          'Chrome executable `$chromeBin` (specified by CHROME_BIN) does not exists.');
+    }
+    final userDataDir = tempDir.createTempSync('user');
```
Comment (jonasfj) [inline pkg/pub_integration/lib/src/headless_env.dart]: let's be `async` when it's easy and cheap :)

File: .travis.yml (hunk 2 above)
Comment (isoos) [reply .travis.yml]: Keeping it simple for now: I've hard-coded the path in the code, and added a TODO to make this better.

**Pull request merged**